includes/server/class-llms-rest-access-plans-controller.php

```diff
@@ -0,0 +1,1081 @@
+					'Determines the price of the trial period.
+					Only applies when `trial_enabled` is `true`.',
+					'lifterlms'
+				),
+				'type'        => 'number',
+				'default'     => 0,
+				'context'     => array( 'view', 'edit' ),
+				'arg_options' => array(
+					'validate_callback' => 'llms_rest_validate_positive_float_w_zero',
+				),
+			),
+			'visibility'                => array(
+				'description' => __(
+					'Access plan visibility.',
+					'lifterlms'
+				),
+				'type'        => 'string',
+				'default'     => 'visible',
+				'enum'        => array_keys( llms_get_access_plan_visibility_options() ),
+				'context'     => array( 'view', 'edit' ),
+			),
+		);
+
+		$schema['properties'] = array_merge(
+			(array) $schema['properties'],
+			$access_plan_properties
+		);
+
+		/**
+		 * Filter item schema for the access-plan controller
+		 *
+		 * @since [version]
+		 *
+		 * @param array $schema Item schema data.
+		 */
+		$schema = apply_filters( 'llms_rest_access_plan_item_schema', $schema );
+
+		return $schema;
+	}
+
+	/**
+	 * Retrieves the query params for the objects collection
+	 *
+	 * @since [version]
+	 *
+	 * @return array Collection parameters.
+	 */
+	public function get_collection_params() {
+
+		$query_params = parent::get_collection_params();
+
+		$query_params['post_id'] = array(
+			'description'       => __( 'Retrieve access plans for a specific list of one or more posts. Accepts a course/membership id or comma separated list of course/membership ids.', 'lifterlms' ),
+			'type'              => 'array',
+			'items'             => array(
+				'type' => 'integer',
+			),
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		return $query_params;
+	}
+
+	/**
+	 * Retrieves an array of arguments for the delete endpoint
+	 *
+	 * @since [version]
+	 *
+	 * @return array Delete endpoint arguments.
+	 */
+	public function get_delete_item_args() {
+		return array();
+	}
+
+	/**
+	 * Whether the delete should be forced
+	 *
+	 * @since [version]
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool True if the delete should be forced, false otherwise.
+	 */
+	protected function is_delete_forced( $request ) {
+		return true;
+	}
+
+	/**
+	 * Whether the trash is supported
+	 *
+	 * @since [version]
+	 *
+	 * @return bool True if the trash is supported, false otherwise.
+	 */
+	protected function is_trash_supported() {
+		return false;
+	}
+
+	/**
+	 * Check if a given request has access to create an item
+	 *
+	 * @since [version]
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function create_item_permissions_check( $request ) {
+
+		$can_create = parent::create_item_permissions_check( $request );
+
+		// If current user cannot create the item because of authorization, check if the current user can edit the "parent" course/membership.
+		if ( llms_rest_is_authorization_required_error( $can_create ) ) {
```

Comment:
I think we can probably skip the parent method check and always rely on this parent (course/membership) post check below.

I think, right? What do we gain by doing both?

What do we gain by doing both checks?

Reply:
Well the idea is that we want to keep _all_ the checks we make on the abstract, which atm aren't that many, but stil that's the idea of abstraction... I think eheh
E.g.
https://github.com/gocodebox/lifterlms-rest/blob/1.0.0-beta.17/includes/abstracts/class-llms-rest-posts-controller.php#L187-L190
If I don't call the parent class method I have to do that check here.
Also, in the parent class method we also check if the current user can create this post type (publish), which for _admins_ is always true: for them we don't need to check if they can edit the parent course/membership.

I know that _practically_ for _admins_ checking if they can publish access plans is equivalent to check if they can edit the parent course, but still what we're doing here is _formally_ more correct.
I'd like to keep this as it is :D
But let me know if you prefer otherwise.